src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -194,15 +203,17 @@
         lifecycle-fns {:shutdown-on-error agent-shutdown-fn
                        :initialize-pool-instance initialize-pool-instance-fn
                        :cleanup cleanup-fn}
-        modified-jruby-config (assoc jruby-config :ruby-load-path (managed-load-path
-                                                                   (:ruby-load-path jruby-config)))]
+        modified-jruby-config (-> jruby-config
+                                  (assoc :ruby-load-path (managed-load-path
+                                                          (:ruby-load-path jruby-config)))
+                                  initialize-gem-path)]
     (jruby-core/initialize-config (assoc modified-jruby-config :lifecycle lifecycle-fns))))
 
 (schema/defn initialize-and-create-jruby-config :- jruby-schemas/JRubyConfig
   "Handles the initialization of the jruby-puppet config for the purpose of returning a
   jruby config. This function will only use the :jruby-puppet and :http-client sections
   from raw-config. If values are not provided, everything in :http-client :jruby-puppet
-  will be given default values, except for :ruby-load-path and :gem-home in the
+  will be given default values, except for :ruby-load-path, :gem-home, and :gem-path in the
```

Comment:
Yeah it may just be that the wording is confusing here.  I think what it's trying to say is that the `jruby-utils` library will merge in defaults for everything except for those.  I'll see if I can't figure out a way to tweak the wording.